js/ftx.js

```diff
@@ -1695,42 +1714,91 @@ module.exports = class ftx extends Exchange {
 
     async fetchOrder (id, symbol = undefined, params = {}) {
         await this.loadMarkets ();
+        const market = (symbol !== undefined) ? this.market (symbol) : undefined;
         const request = {};
         const clientOrderId = this.safeValue2 (params, 'client_order_id', 'clientOrderId');
-        let method = 'privateGetOrdersOrderId';
+        const options = this.safeValue (this.options, 'fetchOrder', {});
+        const defaultMethod = this.safeString (options, 'method', 'privateGetOrdersOrderId');
+        let method = this.safeString (params, 'method', defaultMethod);
         if (clientOrderId === undefined) {
             request['order_id'] = id;
         } else {
             request['client_order_id'] = clientOrderId;
             params = this.omit (params, [ 'client_order_id', 'clientOrderId' ]);
             method = 'privateGetOrdersByClientIdClientOrderId';
         }
-        const response = await this[method] (this.extend (request, params));
+        const type = this.safeValue (params, 'type');
+        params = this.omit (params, 'type');
+        // Note, the below types use non-standard endpoints, look through the implementation to understand better (It might better to use `fetchOrders/fetchOpenOrders` to get your trigger order's details)
+        if ((type === 'stop') || (type === 'trailingStop') || (type === 'takeProfit')) {
+            const orderStatusFlag = this.safeString (params, 'orderStatusFlag');
+            params = this.omit (params, 'orderStatusFlag');
+            if (method === 'privateGetConditionalOrdersConditionalOrderIdTriggers') {
+                const order = await this.fetchConditionalTriggersHelper (id, params);
+                order['trigger_order_flag'] = true;
```

Comment:
This is a no-go, we should not inject this field into the raw order to be parsed. The order must remain untouched, so no injections and no removals. Also, if possible, please, remove the fetchConditionalTriggerHelper, it does not really save us much space, but makes the code harder to follow. Ideally, please solve this task without introducing new methods or fields into the equation.

Reply:
considered all notes, just I've one question:
`fetchConditionalTriggersHelper` is not being used alone on that line, but that is being used from 2 different places (that's why i thought to put in a helper method, as the code was reused. also, i've included special comments in that helper method, to explain the related endpoint's behavior there). should I still remove that helper and place two copies of that code-block (with same comments) into those two places?